### DISCUSSION TITLE
mode definitions with schema

### DISCUSSION THREAD

**Discussion by ahoneycutt** [dagster-io/dagster#3385] [Q&A] [answered]
Hello, I'm trying to create static mode definitions with configured resources. Perhaps I'm using the API incorrectly, but I basically wish to have a dev mode and a prod mode, with resources defined for each. What I'm unsure about is how to pass in the `config_schema`. Here's what I'm attempting to do:
```python
# pipeline mode definition
mode_development = ModeDefinition('dev', resource_defs={
    'service_one': configured(
        http_resource, config_schema={
            'host': String('127.0.0.1'),
            'port': Int(8080),
        }
    ),
    'service_two': configured(
        http_resource, config_schema={
            'host': String('127.0.0.1'),
            'port': Int(8888),
        }
    ),
})

# solid
@solid(required_resource_keys={'service_one'})
def my_solid(context: SolidExecutionContext):
    yield Output(1, output_name='number')
```
The problem is that once we get to the solid, it raises:

```
dagster.check.CheckError: Value in dictionary mismatches expected type for key service_one. Expected value of type <class 'dagster.core.definitions.resource.ResourceDefinition'>. Got value <function configured.<locals>._configured at 0x10d3be8b0> of type <class 'function'>.
```

What's the preferred approach to configuring `resource_defs` in a mode definition?

Thanks!

EDIT: correct usage of String/Int

**Comment by ahoneycutt** [comment, marked as answer]
I have answered my own question. Specifically, a call to the resource's `configured` method is appropriate here. See https://docs.dagster.io/overview/configuration/configured#main for more information.

```python
# pipeline mode definition
mode_development = ModeDefinition('dev', resource_defs={
    'service_one': http_resource.configured({
        'host': String('127.0.0.1'),
        'port': Int(8080),
    }),
    'service_two': http_resource.configured({
        'host': String('127.0.0.1'),
        'port': Int(8888),
    }),
})
```
Edit: Correct usage of String/Int

**Comment by catherinewu** [comment]
Hi @ahoneycutt! 👋  Thanks for sending over this question. We're working on adding "How-to" guides to our docs and I just added this question to the list. We would love to hear feedback you have about dagster or about "How-to" guides you would like to see, either via GH Discussions or on Slack :)
